vsphere/stackstate_checks/vsphere/vsphere.py

```diff
@@ -509,38 +503,57 @@ def _get_server_instance(self, instance):
 
         # If both configs are used, log a message explaining the default
         if not ssl_verify and ssl_capath:
-            self.log.debug("Your configuration is incorrectly attempting to "
-                           "specify both a CA path, and to disable SSL "
-                           "verification. You cannot do both. Proceeding with "
-                           "disabling ssl verification.")
+            self.log.debug(
+                "Your configuration is incorrectly attempting to "
+                "specify both a CA path, and to disable SSL "
+                "verification. You cannot do both. Proceeding with "
+                "disabling ssl verification."
+            )
+
+        try:
+            # Object returned by SmartConnect is a ServerInstance
+            # https://www.vmware.com/support/developer/vc-sdk/visdk2xpubs/ReferenceGuide/vim.ServiceInstance.html
+            server_instance = connect.SmartConnect(
+                host=instance.get('host'),
+                user=instance.get('username'),
+                pwd=instance.get('password'),
+                sslContext=context if not ssl_verify or ssl_capath else None,
```

Comment:
You dropped `connectionPoolTimeout=-1`, will we not run into the same issue if we make a long running requests for which this was the fix?

Reply:
looks like not, the main issue with this kind of problem was only `session` timing out from vCenter Esxi and only way to fix this is to re-login and this is what the fix made right now.

Reply:
Ok just verify this. I remember you did some test scenario last time round to verify this fix worked for the previous customer. Try and run that again and see what happens

Reply:
@zandrewitte yeah, I know but the other customer is also facing this now I think.